src/System.Management.Automation/engine/SessionStateDriveAPIs.cs

```diff
@@ -508,12 +508,13 @@ private PSDriveInfo GetDrive(string name, bool automount)
 
             if (result == null && automount)
             {
-                result = AutomountBuiltInDrive(name);
-            }
-
-            if (result == null && this == ExecutionContext.TopLevelSessionState)
-            {
-                result = AutomountFileSystemDrive(name);
+                // first try to automount as a file system drive
+                result = AutomountFileSystemDrive(name);

```

Comment:
After this change, new drives will be created in global scope; I've added a check for this to tests.